targets/rpc.cpp

```diff
@@ -0,0 +1,450 @@
+    return setup.get();
+}
+}; // namespace
+
+void check_btcd()
+{
+    JSONRPCRequest req = rpc_testing_setup->GenerateRequest("help", {""});
+    int err = rpc_testing_setup->CallRPCBtcd(req);
+    if (err == -2) {
+        std::cerr << "Error: Could not connect to the BTCD server \n";
+        std::cerr << "Please start the btcd server by executing:\n";
+        std::cerr << "   btcd --connect=127.0.0.1 -P bitcoinfuzz -u bitcoinfuzz --notls\n";
+        std::terminate();
+    } else if (err == 401) { // Unauthorized
+        std::cerr << "Error: BTCD server has been started with an unexpected username / password combo.\n";
+        std::cerr << "Please ensure that both the username and password are: bitcoinfuzz\n";
+        std::terminate();
+    }
+}
+
+void initialize_rpc()
+{
+    rpc_testing_setup = InitializeRPCFuzzTestingSetup();
+    SetRPCWarmupFinished();
+    const std::vector<std::string> supported_rpc_commands = rpc_testing_setup->GetRPCCommands();
+    for (const std::string& rpc_command : supported_rpc_commands) {
+        const bool safe_for_fuzzing = std::find(RPC_COMMANDS_SAFE_FOR_FUZZING.begin(), RPC_COMMANDS_SAFE_FOR_FUZZING.end(), rpc_command) != RPC_COMMANDS_SAFE_FOR_FUZZING.end();
+        const bool not_safe_for_fuzzing = std::find(RPC_COMMANDS_NOT_SAFE_FOR_FUZZING.begin(), RPC_COMMANDS_NOT_SAFE_FOR_FUZZING.end(), rpc_command) != RPC_COMMANDS_NOT_SAFE_FOR_FUZZING.end();
+        if (!(safe_for_fuzzing || not_safe_for_fuzzing)) {
+            std::cerr << "Error: RPC command \"" << rpc_command << "\" not found in RPC_COMMANDS_SAFE_FOR_FUZZING or RPC_COMMANDS_NOT_SAFE_FOR_FUZZING. Please update " << __FILE__ << ".\n";
+            std::terminate();
+        }
+        if (safe_for_fuzzing && not_safe_for_fuzzing) {
+            std::cerr << "Error: RPC command \"" << rpc_command << "\" found in *both* RPC_COMMANDS_SAFE_FOR_FUZZING and RPC_COMMANDS_NOT_SAFE_FOR_FUZZING. Please update " << __FILE__ << ".\n";
+            std::terminate();
+        }
+    }
+    const char* limit_to_rpc_command_env = std::getenv("LIMIT_TO_RPC_COMMAND");
+    if (limit_to_rpc_command_env != nullptr) {
+        g_limit_to_rpc_command = std::string{limit_to_rpc_command_env};
+    }
+
+    check_btcd();
+}
+
+FUZZ_TARGET(rpc, .init = initialize_rpc)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    bool good_data{true};
+    SetMockTime(ConsumeTime(fuzzed_data_provider));
+
+    const std::string rpc_command = fuzzed_data_provider.ConsumeRandomLengthString(64);
+    if (!g_limit_to_rpc_command.empty() && rpc_command != g_limit_to_rpc_command) {
+        return;
+    }
+
+    const bool safe_for_fuzzing = std::find(RPC_COMMANDS_SAFE_FOR_FUZZING.begin(), RPC_COMMANDS_SAFE_FOR_FUZZING.end(), rpc_command) != RPC_COMMANDS_SAFE_FOR_FUZZING.end();
+    if (!safe_for_fuzzing) {
+        return;
+    }
+
+    std::vector<std::string> arguments;
+    LIMITED_WHILE(good_data && fuzzed_data_provider.ConsumeBool(), 100)
```

Comment:
Why do we need `good_data`?

Reply:
To not push further to the `arguments` incase `ConsumeRPCArgument` has already failed previously.(Check after lien 280

Reply:
ah ok, I see. Same of Core.